File: ts/src/parkingBoy/parkingBoy.ts
/** @format */

import {ParkingLot} from '../parkingLot/parkingLot'
import {ParkingBoyRule} from './parkingBoyRule'

export class ParkingBoy {
    public constructor(private parkingLots: ParkingLot[], private parkingBoyRule: ParkingBoyRule) {}

    public checkAvailableLot() {
        return this.parkingLots.some(({hasAvailableSpaces}) => hasAvailableSpaces)
    }

    public findOneValidParkingLot() {
        return this.parkingBoyRule.findOneValidParkingLot(this.parkingLots)
    }
}

File: kt/src/main/kotlin/com/github/scglwsj/parking/boy/ParkingBoy.kt
package com.github.scglwsj.parking.boy

import com.github.scglwsj.parking.lot.ParkingLot

class ParkingBoy constructor(
    private val parkingLots: List<ParkingLot>,
    private val parkingBoyRule: ParkingBoyRule
) {
    fun checkAvailableLot(): Boolean = parkingLots.any { it.hasAvailableSpaces }
    fun findOneValidParkingLot(): ParkingLot? = parkingBoyRule.findOneValidParkingLot(parkingLots)
}
